### PR TITLE
fix: fix dataSource not beign required in fetcher

### PR DESCRIPTION
**Pull request by nenadfilipovic**
🚨 **IMPORTANT: Please do not create a Pull Request without creating an issue first.**

*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

## Description

Provide fix, so that dataSource is required param in situations where user have not provided custom endpoint because fetch must get it from somewhere.

#6633

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Run build-and-test with all tests passed (had to update snapshot)

**Test Environment**:
- OS: Linux
- `@graphql-codegen/...`: latest
- NodeJS: 14

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by n1ru4l** [comment]
@nenadfilipovic Can you please add a changeset?

File: .changeset/unlucky-bears-march.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@graphql-codegen/typescript-react-query': minor
```
Comment (n1ru4l) [inline .changeset/unlucky-bears-march.md:2]:
```suggestion
'@graphql-codegen/typescript-react-query': patch
```

**Review by n1ru4l — APPROVED** [review]

**Comment by n1ru4l** [comment]
@nenadfilipovic Awesome, thanks 🔥

**Comment by nenadfilipovic** [comment]
@n1ru4l thanks

**Pull request merged**